library/src/image.cxx

```diff
@@ -93,6 +93,27 @@ class image::internals
       }
     }
   }
+
+  static bool checkSaveFormatCompatibility(const image& self, SaveFormat format)
```

Comment:
How about making this a `void` method that throws so the message can be more precise? e.g. "PNG only supports channel type BYTE and SHORT", "JPG only supports channel type BYTE" and so on.